Queueomatic/Server/Endpoints/Participant/Delete/DeleteParticipantEndpoint.cs

```diff
@@ -20,7 +22,38 @@ public override void Configure()
 
     public override async Task HandleAsync(DeleteParticipantRequest req, CancellationToken ct)
     {
+        if (req.ParticipantId != null && 
+            req.ParticipantId != req.Id)
+        {
+            await SendUnauthorizedAsync();
+            return;
+        }
+
+        if (!string.IsNullOrEmpty(req.UserId) && 
+            !await IsUserOwnerOfTheRoom(req))
+        {
+            await SendUnauthorizedAsync();
+            return;
+        }
+        
         await _participantService.DeleteOneAsync(req.Id);
         await SendNoContentAsync();
     }
+
+    private async Task<bool> IsUserOwnerOfTheRoom(DeleteParticipantRequest req)
+    {
+        var user = await _unitOfWork.UserRepository.GetAsync(req.UserId);
+        if (user == null) return false;
+
+        var userRooms = user.Rooms.ToList();
+        
+        var isUserTheOwner =
```

Comment:
This naming is not representative of what is being returned.

If you want to check if a user owns a room, you can use [Any()](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.any?view=net-7.0) instead

Reply:
Oh useless/dummy variable. Fixed.

Reply:
Are you sure it is fixed? I was referring to line 48. The naming scheme you use indicates that a bool is being returned from the expression, but you are using FirstOrDefault, which returns an object. Either change the name of the variable or update the code with Any(). 

Any() is also performance wise better since it does not retrieve the object and allocates memory for it,

Reply:
And also changed the name.